WIP testserver: add a unique message whenever starting a test server

yuzefovich opened the pull request:
Epic: None

yuzefovich committed:
testserver: add a unique message whenever starting a test server

The string "considering starting a default test tenant" will uniquely
identify the top-level unit test that started at least one test server.
Note that it's possible that a single unit test starts servers multiple
times (for each subtest), each subtest will get a separate message. This
seems ok.

Release note: None

yuzefovich committed:
WIP on always picking shared process during randomization

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/115149)
<!-- Reviewable:end -->

yuzefovich committed:
testserver: more decision messages

Add
- "didn't start default test tenant due to test specifying explicit decision"
- "default test tenant is always enabled" log messages.
- "cluster virtualization disabled for benchmarks"
- "default test tenant is disabled by environment"
- "cluster virtualization disabled with no issue number"
- "cluster virtualization disabled in global scope with no issue"
